lib/manager/pipenv/extract.js

```diff
@@ -43,11 +44,28 @@ function extractFromSection(pipfile, section, registryUrls) {
     return [];
   }
   const specifierRegex = new RegExp(`^${specifierPattern}$`);
-  const deps = Object.entries(pipfile[section])
+  const pipfileSection = pipfile[section];
+
+  Object.keys(pipfileSection).forEach(key => {
+    if (is.object(pipfileSection[key]))
+      pipfileSection[key].version = pipfileSection[key].version || '*';
+  });
+
+  const deps = Object.entries(pipfileSection)
     .map(x => {
       const [depName, requirements] = x;
       let currentValue;
       let pipenvNestedVersion;
+      if ([requirements, requirements.version].find(r => r === '*')) {
```

Comment:
Couldn't this section now be replaced with a check near the bottom like this? `if (dep.currentValue === '*') dep.skipReason = 'any-version';` ?

Reply:
This wouldn't work.
![ifail](https://user-images.githubusercontent.com/22542906/56416848-c8829f00-629a-11e9-9314-9417c6e0a5e6.png)

Cannot read property `deps` of NULL

![return-null](https://user-images.githubusercontent.com/22542906/56416855-cf111680-629a-11e9-869f-92254e9ce07a.png)